### PR TITLE
vscode: filter out arm linux from using prebuilt binaries

### LINKED ISSUE #3076: ARM linux downloads an x86 plugin that does not work (vscode on Jetson Nano - probably same on Raspbery Pi)
This worked on ARM devices until recent changes. It is not clear what changed in the build process that it needs to download even when building from source (both Nano and Macos).

Perhaps there is a work around? Macos works but I had to cancel past a retro-frightening "Need to install Java" popup. Java is already on the Nano (Ubuntu).

### PR DESCRIPTION
**Pull request by Veetaha**
Closes #3076

### CODE REVIEW

**Comment by Veetaha** [comment]
@paulirotta, can you please take a look? I am not en expert in platform-dependency...

File: editors/code/src/config.ts [hunk 1]
```diff
@@ -64,9 +64,24 @@ export class Config {
...
+
         switch (platform) {
-            case "linux":  return "ra_lsp_server-linux";
+
+            case "linux": {
+                switch (arch) {
+                    case "arm":
+                    case "arm64": return null;
+
+                    default: return "ra_lsp_server-linux";
```
Comment (lnicola) [inline editors/code/src/config.ts]: I would reverse this check: return `ra_lsp_server-linux` for `x64` and `null` in the other cases.

Same for Windows (`x64`, @matklad can confirm). It's probably not realistic to offset 32-bit builds at this time.

Comment (Veetaha) [reply editors/code/src/config.ts]: Okay, thank you for the hint. And what about Mac?

**Comment by matklad** [comment]
I guess OS and arch are orthogonal dimensions. We need one check: "if arch
!= x64 { error!("sorry Dave, I can't do that")}"

File: editors/code/src/config.ts (hunk 1 above)
Comment (lnicola) [reply editors/code/src/config.ts]: Same thing. I _assume_ it returns `x64`, but I can't test.

**Comment by lnicola** [comment]
> I guess OS and arch are orthogonal dimensions.

Well, they're not orthogonal since both part of the target triple/tuple. But RA uses quite a lot of memory and it probably won't work on 32-bit platforms, while it does work on e.g. ARM, but we can't build it on CI.

**Comment by Veetaha** [comment]
Can't we do cross-compilation on CI?

**Comment by lnicola** [comment]
We can or might, but we don't have it today. Anyway, it seems more feasible than 32-bit or not-x86 support.

**Comment by jminer** [comment]
I usually use a 32-bit build of RA, and it works well enough. On one project for a period of time, RA was getting OOM sometimes, but it uses less memory now. The only reason is that I normally target 32-bit Windows, so that's the only toolchain I have installed, and I build RA with it too. However, for prebuilt binaries, I doubt there's any demand for 32-bit Windows. Maybe some people would use it on 32-bit Arm? I don't know if it would ever be worth offering prebuilt binaries though.

**Comment by lnicola** [comment]
I've seen people using a Raspberry Pi for development and somebody using one as their main computer. Someone else here seemed to be using a Jetson (AArch64?). FreeBSD binaries would also be nice, I suppose.

**Pull request merged**